src/us/kbase/narrativemethodstore/db/mongo/MongoDynamicRepoDB.java

```diff
@@ -182,24 +189,26 @@ public void registerRepo(String userId, RepoProvider repoDetails)
                         "can not be updated by non-git repository: [" + oldUrl + 
                         "] -> [" + newUrl + "]");
         }
-        RepoData repoData = JsonRepoProvider.repoProviderToData(this, repoDetails);
-        MongoCollection hist = jdb.getCollection(TABLE_REPO_HISTORY);
-        hist.insert(String.format("{%s:#,%s:#,%s:#}", FIELD_RH_MODULE_NAME,
-                FIELD_RH_VERSION, FIELD_RH_REPO_DATA), repoModuleName,
-                newVersion, repoData);
-        MongoCollection data = jdb.getCollection(TABLE_REPO_INFO);
+        final RepoData repoData = JsonRepoProvider.repoProviderToData(this, repoDetails);
+        final DBCollection hist = db.getCollection(TABLE_REPO_HISTORY);
+        hist.insert(new BasicDBObject(FIELD_RH_MODULE_NAME, repoModuleName)
+                .append(FIELD_RH_VERSION, newVersion)
+                .append(FIELD_RH_REPO_DATA, toDBObject(repoData)));
+        
+        final DBCollection data = db.getCollection(TABLE_REPO_INFO);
+        //should just do an upsert rather than handling the logic application side
```

Comment:
I know you are trying to minimize changes, but wouldn't the upsert be easier to write too?

Reply:
Well, not for me - the easier thing is to leave the logic alone and just do the conversion, as I don't need to think about the logic.